apache/metron
# METRON-1423: Ambari work to handle Solr configuration

## Contributor Comments
This PR allows support for Metron-related Solr configuration and management through Ambari.  That does NOT include managing the actual Solr service.  For now this is done outside of Ambari.

High level changes include:

- Solr is now installed automatically in full dev, although not in a running state.  This implementation was inspired by the `install_solr.sh` script.
- Additional scripts for managing the Solr installation provided in full dev and installed with `install_solr.sh`.
- Exposes a single Zookeeper Solr configuration (for now) parameter through Ambari.
- Replaced the random access index writer class Ambari parameter with a random access search engine parameter that is a drop down of "Elasticsearch" and "Solr".  
- Replaced "Elasticsearch" with "Random Access" in all the Ambari parameter descriptions.
- Added the Solr schema creation to Ambari on Indexing start.  This matches the ES template loading implementation with the exception of exposing it through Service actions.

Testing instructions are as follows:
1. Spin up full dev
2. Shutdown ES and Kibana
3. Start Solr from the command line with $METRON_HOME/bin/start_solr.sh
4. Navigate to the Indexing tab in Ambari > Metron > Configs and change the "Random Access Search Engine" setting from "Elasticsearch" to "Solr"
5. Now you can start ingesting data into Solr in 2 ways.  After you make the change in Ambari it will prompt you to restart affected services.  A restart does not automatically create collections so you will need to do this outside of Ambari for bro, snort and error with $METRON_HOME/bin/create_collection.sh (first argument is sensor name) before you restart.
6. Or you can simply stop Metron Indexing and then start it.  This will trigger Ambari to automatically load the collections.
7. Navigate to http://node1:8983/solr/#/snort/query or http://node1:8983/solr/#/bro/query and hit the "Execute Search" button.  You should see data.

There is a lot to digest here and still more questions to answer but this should get us started.  Outstanding items include:

- What Solr parameters do we expose in Ambari?  All of them?
- Do we want to add Solr collection functions to Service actions?  I don't see a way to make this conditional on the selected search engine so both options would always be shown
- Need to review and add documentation
- Support for REST configuration (probably a separate PR)

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:


### For code changes:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```


### For documentation related changes:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.
